src/suggestion-box.ts

```diff
@@ -0,0 +1,383 @@
+            for (let key of Object.keys(obj)) {
+                let template: any;
+                if (key === "name") {
+                    template =
+                        this.highlight === "matched"
+                            ? this._highlight(obj[key])
+                            : this._startsWithHighlight(obj[key]);
+                } else {
+                    template = obj[key];
+                }
+
+                str = str.replace(
+                    new RegExp("\\{" + key + "\\}", "gi"),
+                    template
+                );
+            }
+            return str;
+        }
+    }
+    keyDownInputEvent(e: any) {
+        const resultLength = this.result.length;
+        switch (e.key) {
+            case "ArrowUp":
+                this.index =
+                    this.index === -1 || (this.has_more && this.index === 0)
+                        ? resultLength + (this.has_more ? 0 : 0)
+                        : (this.index -
+                              1 +
+                              resultLength +
+                              (this.has_more ? 1 : 0)) %
+                          (resultLength + (this.has_more ? 1 : 0));
+
+                this.setActive(resultLength);
+                break;
+            case "ArrowDown":
+                this.index =
+                    this.index === -1 ||
+                    (this.has_more && this.index === resultLength)
+                        ? 0
+                        : (this.index + 1) %
+                          (resultLength + (this.has_more ? 1 : 0));
+                this.setActive(resultLength);
+                break;
+
+            case "Enter":
+                if (this.index === resultLength) {
+                    this.loadmoreentry(e);
+                } else {
+                    this.setInputValue();
+                }
+                break;
+        }
+    }
+    InputEvent(e: any) {
+        this.prevIndex = 0;
+        this.index = -1;
+        this.input_el = e.target;
+        if (this.list) {
+            if ((this.input_el as HTMLInputElement)?.value) {
+                this.filtered_list = this.list.filter((item) =>
+                    item[this.listkey].includes(
+                        (this.input_el as HTMLInputElement)?.value
+                    )
+                );
+
+                this.setSuggestion(this.filtered_list, false);
+            } else {
+                this.filtered_list = [];
+
+                this.setSuggestion(this.filtered_list, false);
+            }
+        }
+        return;
+    }
+
+    setActive(resultLength: number) {
+        if (this.prevIndex >= 0 && this.prevIndex < resultLength) {
+            const prevItem = this.private_items[this.prevIndex];
+            if (prevItem) {
+                prevItem.classList.remove("sgn-active");
+            }
+        }
+        if (this.index >= 0 && this.index < resultLength) {
+            const selectedItem = this.private_items[this.index];
```

Comment:
This entire if block can be replaced to just one line
this.private_items[this.index]?.classList.add("sgn-active")